enclave-runtime/src/sidechain_impl.rs

```diff
@@ -215,20 +231,23 @@ where
 /// Implements `BlockImport`. This is not the definite version. This might change depending on the
 /// implementation of #423: https://github.com/integritee-network/worker/issues/423
 #[derive(Clone)]
-pub struct BlockImporter<A, PB, SB, O, ST, StateHandler> {
+pub struct BlockImporter<A, PB, SB, O, ST, StateHandler, StateKey> {
 	state_handler: Arc<StateHandler>,
+	state_key: StateKey,
 	_phantom: PhantomData<(A, PB, SB, ST, O)>,
 }
 
-impl<A, PB, SB, O, ST, StateHandler> BlockImporter<A, PB, SB, O, ST, StateHandler> {
+impl<A, PB, SB, O, ST, StateHandler, StateKey>
+	BlockImporter<A, PB, SB, O, ST, StateHandler, StateKey>
+{
 	#[allow(unused)]
-	pub fn new(state_handler: Arc<StateHandler>) -> Self {
-		Self { state_handler, _phantom: Default::default() }
+	pub fn new(state_handler: Arc<StateHandler>, state_key: StateKey) -> Self {
+		Self { state_handler, state_key, _phantom: Default::default() }
```

Comment:
This was more of a small improvement on the side: The block importer should not read the state encryption key directly from file, but rather have it passed as member when it's constructed. This makes the dependency more obvious and lets us test this importer more easily, without having to rely on a file existing on the filesystem.